### PR TITLE
COMMERCE-11276 Price on Application on Price Lists & Promotions

### PR DESCRIPTION
**Pull request by smottal**
https://issues.liferay.com/browse/COMMERCE-11276

### CODE REVIEW

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/portlet/action/EditCommercePriceEntryMVCActionCommand.java [hunk 1]
```diff
@@ -222,16 +217,36 @@ protected CommercePriceEntry updateCommercePriceEntry(
 		boolean neverExpire = ParamUtil.getBoolean(
 			actionRequest, "neverExpire");
 
-		ServiceContext serviceContext = ServiceContextFactory.getInstance(
-			CommercePriceEntry.class.getName(), actionRequest);
+		boolean overrideDiscount = ParamUtil.getBoolean(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/portlet/action/EditCommercePriceEntryMVCActionCommand.java:220]: move to 2nd position to sort variable as used

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/price/CommerceProductPriceCalculationV2Impl.java [hunk 2]
```diff
@@ -120,23 +120,37 @@ public CommerceProductPrice getCommerceProductPrice(
...
+				 CommerceBigDecimalUtil.lte(
+					 promoPriceCommerceMoney.getPrice(),
+					 unitPriceCommerceMoney.getPrice()))) {
+
+				commercePriceListId = commercePromoPriceListId;
+				finalPrice = promoPriceCommerceMoney.getPrice();
+				priceOnApplication =
+					promoPriceCommerceMoney.isPriceOnApplication();
+			}
+			else {
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/price/CommerceProductPriceCalculationV2Impl.java:150]: @luca-pellizzon can you please check this?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Please add check to the API too, and manage the checkout button with this story.

Thank you!

**Comment by smottal** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/portlet/action/EditCommercePriceEntryMVCActionCommand.java (hunk 1 above)
Comment (smottal) [reply modules/apps/commerce/commerce-pricing-web/src/main/java/com/liferay/commerce/pricing/web/internal/portlet/action/EditCommercePriceEntryMVCActionCommand.java]: Done

**Comment by smottal** [comment]
Prevent checkout done in [this commit](https://github.com/liferay-commerce/liferay-portal/pull/3841/commits/49b04e428393c83b0fcc3e62c46c4fa01ec88bed).

According to the task the API control is no longer needed

**Review by andrea-ale-sbarra — APPROVED** [review]
LGTM.

Thank you!

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by smottal** [comment]
ci:test:sf

**Pull request closed without merging**